### PR TITLE
Filter: Menu Items 

### LINKED ISSUE #83: Filter by Menu Ingredient
Add drop down for users to filter Menu Items by Ingredients

# User Story
When a user navigates to Menu Items, there should be a drop down to filter by available ingredients. 

# Acceptance Criteria
* There should be a dropdown with checkboxes for each available ingredient 
* When a user selects multiple ingredients and submits, only menu items that contains all of those ingredients should appear on the web app. 

# Dependecies
- [X] A dropdown needs to be added to show all ingredients

# Dev Notes
- [ ] Capture all selected checkboxes and store in array
- [ ] Pass that array to a promise that will filter
- [ ] Make a call to Firebase, compare the ingredient array to menu item ingredients array to return only menu items that contain all ingredients in the passed in array

### PR DESCRIPTION
**Pull request by caseywalker**
New Feature: Filter Menu Items by ingredients

## Description
* A user may now navigate to Menu Items, click 'Filter by Ingredients', select a list of ingredients, and filter down to Menu Items that contain all selected ingredients. 

## Related Issue
Fixes #83 

## Motivation and Context
* We have a very robust menu, and we want our users to be able view our website and see if we offer menu items that meet their exact cravings by available ingredients. 
* Also require new menu items to have at least 1 ingredient added to prevent any bugs on the filter. 

## How Can This Be Tested?
1) Pull down cw-filter-menu, npm start, navigate to 'Menu Items' Click 'Filter by Ingredients' 
2) Select a few ingredients, and then hit the 'Submit' button, and see which menu items match the results. 

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/javascripts/events/domEvents.js [hunk 1]
```diff
@@ -19,14 +19,16 @@ import {
...
+import { getMenuIngredients } from '../helpers/data/menuIngredientsData';
 import menuIngredients from '../components/forms/ingredientModal';
 import addReservationForm from '../components/forms/addReservationForm';
 import addStaffForm from '../components/forms/addStaffForm';
 import editIngredientForm from '../components/forms/editIngredientForm';
 import updateStaffForm from '../components/forms/updateStaffForm';
 import editReservationForm from '../components/forms/editReservationForm';
 import editMenuItemForm from '../components/forms/editMenuItems';
+import filterSubmit from '../components/menu/filterSubmit';
+// import filterMenu from '../components/menu/filterMenu';
```
Comment (elliewalkersm) [inline src/javascripts/events/domEvents.js]: Commented out import

File: src/javascripts/events/domEvents.js [hunk 2]
```diff
@@ -91,6 +93,8 @@ const domEventListeners = (e) => {
     formModal('Ingredients');
   }
 
+  // SHOW FILTER MENU ITEMS DROPDOWN
```
Comment (elliewalkersm) [inline src/javascripts/events/domEvents.js]: Is this needed anymore?

**Review by elliewalkersm — DISMISSED** [review]
This code may also nee to be added to the edit menu items form:

    if (itemObject.ingredients.length < 1) {
      // eslint-disable-next-line no-alert
      window.confirm('Please add at least 1 Ingredient');
    }

**Review by elliewalkersm — APPROVED** [review]
looks great!

**Review by seancrossettie — APPROVED** [review]

**Pull request merged**